database/buildSchema/43_views_functions_triggers.sql

```diff
@@ -1516,6 +1516,74 @@ GROUP BY
     3,
     4;
 
+CREATE TABLE IF NOT EXISTS allowed_self_assignable_sections_shape (
+    review_assignment_id int,
+    allowed_sections varchar[]
+);
+-- For each active review_assignment for user we need to know how many sections they are allowed to review
+-- this becomes less then trivial because when allowed_sections is null it means that all sections can be
+-- reviewed by the user, thus we need to deduce how many sections there are total in template and replace
+-- allowed sections with all sections in template (when allowed sections is null)
+CREATE OR REPLACE FUNCTION public.allowed_self_assignable_sections (userId int)
+      RETURNS SETOF allowed_self_assignable_sections_shape
+    AS $$
+   SELECT 
+    review_assignment.id as review_assignment_id,
+    CASE WHEN allowed_sections IS NULL THEN
+        array_agg(distinct(template_section.code))
+    ELSE
+        allowed_sections
+    END AS allowed_sections
+    FROM review_assignment
+        JOIN application ON review_assignment.application_id = application.id
+        JOIN template ON application.template_id = template.id
+        JOIN template_section ON template_section.template_id = template.id
+    WHERE review_assignment.status = 'AVAILABLE' 
```

Comment:
pre filtered by the same filters that are also in the CASE WHEN clause, similar with the method below